ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            do
+            {
+                bytes_per_camp = 0;
+                fseek_function(
+                    pf,
+                    32 + (MM_FILE_OFFSET)pMMBDXP->nFields * 32 +
+                        (MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF + 1 + 4),
+                    SEEK_SET);
+                if (1 != fread_function(&bytes_per_camp, 1, 1, pf) ||
+                    1 != fread_function(&un_byte, 1, 1, pf) ||
+                    1 != fread_function(&tretze_bytes,
+                                        3 + sizeof(bytes_per_camp), 1, pf))
+                {
+                    free(pMMBDXP->pField);
+                    fclose_function(pf);
+                    return 1;
+                }
+                if (bytes_per_camp == 0)
+                    memcpy(&bytes_per_camp, (char *)(&tretze_bytes) + 3,
+                           sizeof(bytes_per_camp));
+
+                bytes_acumulats += bytes_per_camp;
+                pMMBDXP->nFields++;
+            } while (bytes_acumulats < pMMBDXP->BytesPerRecord);
+        }
+    }
+
+    if (pMMBDXP->nFields != 0)
+    {
+        pMMBDXP->pField = MM_CreateAllFields(pMMBDXP->nFields);
+        if (!pMMBDXP->pField)
+        {
+            fclose_function(pf);
+            return 1;
+        }
+    }
+    else
+        pMMBDXP->pField = nullptr;
+
+    fseek_function(pf, 32, SEEK_SET);
+    for (nIField = 0; nIField < pMMBDXP->nFields; nIField++)
+    {
+        if (1 != fread_function(pMMBDXP->pField[nIField].FieldName,
+                                MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF, 1, pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].FieldType), 1, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].reserved_1), 4, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].BytesPerField), 1, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].DecimalsIfFloat), 1,
+                                1, pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].reserved_2), 13, 1,
+                                pf) ||
+            1 != fread_function(&(pMMBDXP->pField[nIField].MDX_field_flag), 1,
+                                1, pf))
+        {
+            free(pMMBDXP->pField);
+            fclose_function(pf);
+            return 1;
+        }
+
+        if (pMMBDXP->pField[nIField].FieldType == 'F')
+            pMMBDXP->pField[nIField].FieldType = 'N';
+
+        pMMBDXP->pField[nIField]
+            .FieldName[MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF - 1] = '\0';
+        if (EQUAL(pMMBDXP->pField[nIField].FieldName,
+                  szMMNomCampIdGraficDefecte))
+            pMMBDXP->IdGraficField = nIField;
+
+        if (pMMBDXP->pField[nIField].BytesPerField == 0)
+        {
+            if (!MM_ES_DBF_ESTESA(pMMBDXP->dbf_version))
+            {
+                free(pMMBDXP->pField);
```

Comment:
is it necessary to do that partial cleanup of pMMBDXP here ? If so, after the free, pField should be likely set to nullptr and nFields to 0

Reply:
Ups, I'ts been missing. For sure it has to be = nullptr.
And the free is a free_function() to be coherent with allocation

It could be freed on destroying but it's safer here.

free_function(pMMBDXP->pField);
pMMBDXP->pField = nullptr;